src/sync/sync_permission.cpp

```diff
@@ -61,6 +61,38 @@ Permission::AccessLevel extract_access_level(Object& permission, CppContext& con
 
 }
 
+// MARK: - Permission
+
+std::string Permission::description_for_access_level(AccessLevel level)
+{
+    switch (level) {
+        case AccessLevel::None: return "none";
+        case AccessLevel::Read: return "read";
+        case AccessLevel::Write: return "write";
+        case AccessLevel::Admin: return "admin";
+    }
+    REALM_UNREACHABLE();
+}
+
+bool Permission::paths_are_equivalent(std::string path_1, std::string path_2,
```

Comment:
I briefly chatted with @kspangsege and updated the code to reflect what we believe are the correct semantics. (I can also make the check more restrictive, to only support a leading `/~/`, if you prefer.)